js/civicrm_stripe.js

```diff
@@ -151,14 +151,12 @@
       // Hide payment if total is 0 and no more participants.
       if ($('#priceset').length) {
         additionalParticipants = cj("#additional_participants").val();
-        // The currentTotal is already being calculated in Form/Contribution/Main.tpl.
-        if(typeof currentTotal !== 'undefined') {
-          if (currentTotal == 0 && !additionalParticipants) {
-            // This is also hit when "Going back", but we already have stripe_token.
-            debugging('ozlkf');
-            // This should not happen on Confirm Contribution, but seems to on 4.6 for some reason.
-            //return true;
-          }
+        currentTotal = $('#pricevalue').data('raw-total');
```

Comment:
currentTotal is used till CiviCRM version 4.6.x. In 4.7 currentTotal is never set and hence the code tries to validate payment details which should not be in case total amount is 0. An alternate method which will work for all version of civicrm is to retrieve data stored in pricevalue.